app/models/katello/pool.rb

```diff
@@ -63,6 +63,10 @@ def products
       self.subscription.products if self.subscription
     end
 
+    def upstream?
+      subscription.redhat?
```

Comment:
does subscription.redhat? return true if it is a red hat sub? If so, then wouldn't this function return true if it is a downstream subscription? Or am I mixing something up in ruby land?

Reply:
OR... always another or.. in this case does "upstream" refer to subscriptions that reside on the RH end? If that is the case, then what "stream" are custom subs referred to?

Reply:
'upstream' here means that it comes from the RHSM portal. AKA, a Red Hat subscription, not a custom or virt_who subscription.

Reply:
ahhh ok. i was thinking in the context of our development where upstream != redhat. thanks @akofink!